src/commands/inhouse.js

```diff
@@ -0,0 +1,266 @@
+                let data = '';
+                resp.on('data', (chunk) => {
+                    data += chunk;
+                    });
+                resp.on('end', () => {
+                    try {
+                        var jsonResp = JSON.parse(data);
+                        _summonerId = jsonResp.id;
+                        resolve(_summonerId);
+                    } catch(e) {
+                        reject(e.message);
+                    }
+                    });
+            }).on("error", (err) => {
+                reject(`Got error: ${err.message}`);
+            });
+        });
+    }
+    
+    getRankWeight(summId) {
+        return new Promise((resolve, reject) => {
+            var _rankWeight = 0;
+            var tier;
+            var division;
+    
+            let fullUrl = urlSummRank + summId + urlFinish;
+            https.get(fullUrl, (resp) => {
+                let data = '';
+                resp.on('data', (chunk) => {
+                    data += chunk;
+                  });
+                resp.on('end', () => {
+                    try {
+                        var jsonResp = JSON.parse(data)
+                        var i;
+                        for(i = 0; i < 3; i++) {
+                            if(jsonResp[i] != null) {
+                                if(jsonResp[i].queueType === 'RANKED_SOLO_5x5') {
+                                    tier = jsonResp[i].tier;
+                                    division = jsonResp[i].rank;
+                                    break;
+                                }
+                            }
+                        }
+                        _rankWeight = utils.calculateSummonerRankWeight(tier, division);
+                        resolve(_rankWeight);
+                    } catch(e) {
+                        reject(e.message);
+                    }
+                  });
+            }).on("error", (err) => {
+                reject(`Got error: ${err.message}`);
+            });
+        });
+       
+    }
+}
+
+class Match {
+    constructor(players, _type) {
+        if(players.length != 10) {
+            console.error("need 10 players");
+            return;
+        }
+
+        this.team1 = [];
+        this.team2 = [];
+        this.type = _type;
+
+        this.sortTeams(players);
+    }
+
+    sortTeams(players) {
+        if(this.type === '4fun') {
+            players.forEach(element => {
+                let teamAssignment = Math.random(1, 3);
+                if(teamAssignment === 1) {
+                    if(this.team1.length < 5) {
+                        this.team1.push(element);
+                    } else {
+                        this.team2.push(element);
+                    }
+                } else {
+                    if(this.team2.length < 5) {
+                        this.team2.push(element);
+                    } else {
+                        this.team1.push(element);
+                    }
+                }
```

Comment:
logic def correct i just don't like it when you have to push the element and manually create the array/ One way i can think of to simplify is 1) shuffle the array first 2) slice the array in half. There is a npm package that can do this https://www.npmjs.com/package/shuffle-array#targetText=shuffle(arr%2C%20%5Boptions%5D),elements%20in%20a%20given%20array%20.&targetText=%5B%20options.copy%20%5D%20%7BBoolean,a%20custom%20random%20number%20generator. ( dont know if its too try hard to add package for this), if we don't wanna add package, take a look at https://javascript.info/task/shuffle#targetText=function%20shuffle%20(%20array%20)%20%7B%20array,sorting%20function%20reorders%20elements%20randomly for simple js function. Slicing is ez just use the array.slice(start, end)

Reply:
also found with dis you dont have to manually check the length everytime! if u have less than 10 ppl